### PR TITLE
Setup GH Actions for tests, releases and guide

### PR DESCRIPTION
**Pull request by mlange-42**
Partial fix for #56

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - uses: actions/checkout@v2
+      - name: Get tag
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+        shell: bash
+      - name: Build
+        run: |
+          cargo build --release
+      - name: Compress
+        run: |
+          cp -f target/release/$REPO${{ matrix.bin_extension }} . > /dev/null
```
Comment (adamreichold) [inline .github/workflows/release.yml:38]: Why sinkhole stdout here?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          tar -czf $REPO-${{ env.RELEASE_VERSION }}-${{ matrix.os }}.tar.gz $REPO${{ matrix.bin_extension }}
```
[the remark this answers is not in the document]
Comment (adamreichold) [reply .github/workflows/release.yml:39]: Did you mean `os_name` instead of `os`?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  push:
+    branches: [ master, dev ]
+  pull_request:
+    branches: [ master, dev ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check:
```
Comment (adamreichold) [inline .github/workflows/tests.yml:13]: Running `check` is somewhat redundant to running `clippy` which does everything `check` does and then some.

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mlange-42) [reply .github/workflows/release.yml:39]: it does not matter, as it is renamed to `asset_name` during the upload

[the remark this answers is not in the document]
Comment (mlange-42) [reply .github/workflows/release.yml]: Still changed it...

[the remark this answers is not in the document]
Comment (adamreichold) [reply .github/workflows/release.yml]: If it renamed anyway, could you just call it `release.tar.gz`?

[the remark this answers is not in the document]
Comment (mlange-42) [reply .github/workflows/release.yml]: Right, why not... 😉

[the remark this answers is not in the document]
Comment (mlange-42) [reply .github/workflows/release.yml]: Done in #157

**Pull request merged**